### PR TITLE
Achapman/ch15705/trading api improvements

### PR DESCRIPTION
**Pull request by nuevoalex**
- Simplify API by removing unused (in our code base) and redundant methods
 - Simplify / fix the fill order loop in trading
 - Add ability to ignore owned shares when creating / filling / trading
 - Add ability to trade in terms of total cost instead of shares
 - fxpAmount -> amount
 - displayPrice -> price

### CODE REVIEW

File: source/contracts/trading/Trade.sol [hunk 1]
```diff
@@ -15,133 +15,105 @@ import 'libraries/CashAutoConverter.sol';
...
+        while (_orderId != 0 && _bestAmount > 0 && _tradeData.loopLimit > 0 && isMatch(_orderId, _type, _orderPrice, _tradeData.price)) {
+            bytes32 _nextOrderId = _orders.getWorseOrderId(_orderId);
+            _orders.setPrice(_tradeData.market, _tradeData.outcome, _orderPrice);
+            _bestAmount = IFillOrder(controller.lookup("FillOrder")).fillOrder(_tradeData.sender, _orderId, _bestAmount, _tradeData.tradeGroupId, _tradeData.ignoreShares);
+            _orderId = _nextOrderId;
+            _orderPrice = _orders.getPrice(_orderId);
+            _tradeData.loopLimit -= 1;
         }
-        if (_orderId != 0) {
+        if (_orderId != 0 && isMatch(_orderId, _type, _orderPrice, _tradeData.price)) {
```
Comment (adrake33) [inline source/contracts/trading/Trade.sol]: This seems a little redundant since `isMatch` already returns false if `_orderId` is 0. Was this intentional?

**Comment by adrake33** [comment]
Looks like Travis is also complaining about several unused function params in Trade.sol.

File: source/contracts/trading/Trade.sol (hunk 1 above)
Comment (nuevoalex) [reply source/contracts/trading/Trade.sol]: Good point, can just be `isMatch`!

**Comment by adrake33** [comment]
There are some parameter mismatches in TestFixture.ts that still need to be resolved.

**Pull request merged**